lib/SPIRV/SPIRVUtil.cpp

```diff
@@ -676,6 +667,118 @@ bool hasArrayArg(Function *F) {
   return false;
 }
 
+void getParameterTypes(Function *F, SmallVectorImpl<StructType *> &ArgTys) {
+  // Start by filling in a skeleton of information we can get from the LLVM type
+  // itself.
+  ArgTys.clear();
+  auto *FT = F->getFunctionType();
+  ArgTys.reserve(FT->getNumParams());
+  bool HasSret = false;
+  for (Argument &Arg : F->args()) {
+    if (!Arg.getType()->isPointerTy())
+      ArgTys.push_back(nullptr);
+    else if (Type *Ty = Arg.getParamStructRetType()) {
+      assert(!HasSret && &Arg == F->getArg(0) &&
+             "sret parameter should only appear on the first argument");
+      HasSret = true;
+      ArgTys.push_back(dyn_cast<StructType>(Ty));
+    } else {
+      ArgTys.push_back(nullptr);
+    }
+  }
+
+  // If there's no mangled name, we can't do anything. Also, if there's no
+  // parameters, do nothing.
+  StringRef Name = F->getName();
+  if (!Name.startswith("_Z") || ArgTys.empty())
+    return;
+
+  Module *M = F->getParent();
+
+  // Skip the first argument if it's an sret parameter--this would be an
+  // implicit parameter not recognized as part of the function parameters.
+  auto *ArgIter = ArgTys.begin();
+  if (HasSret)
+    ++ArgIter;
```

Comment:
Given the fact, that it can be not only the second parameter should this block be adjusted?